Assets/Plugins/LeapMotion/Core/Editor/LeapServiceProviderEditor.cs

```diff
@@ -61,79 +80,205 @@ public class LeapServiceProviderEditor : CustomEditorBase<LeapServiceProvider> {
     }
 
     public override void OnInspectorGUI() {
-      #if UNITY_2019_3_OR_NEWER
+
+#if UNITY_2019_3_OR_NEWER
       // Easily tracking VR-enabled-or-not requires an XR package installed, so remove this warning for now.
-      #else
+#else
       if (UnityEditor.PlayerSettings.virtualRealitySupported && !isVRProvider) {
         EditorGUILayout.HelpBox(
           "VR support is enabled. If your Leap is mounted to your headset, you should be "
           + "using LeapXRServiceProvider instead of LeapServiceProvider. (If your Leap "
           + "is not mounted to your headset, you can safely ignore this warning.)",
           MessageType.Warning);
       }
-      #endif
+#endif
 
       base.OnInspectorGUI();
     }
 
     public virtual void OnSceneGUI() {
-      Vector3 origin = target.transform.TransformPoint(controllerOffset);
 
-      Vector3 local_top_left, top_left, local_top_right, top_right,
-              local_bottom_left, bottom_left, local_bottom_right, bottom_right;
-      getLocalGlobalPoint(-1, 1, 1, out local_top_left, out top_left);
-      getLocalGlobalPoint(1, 1, 1, out local_top_right, out top_right);
-      getLocalGlobalPoint(-1, 1, -1, out local_bottom_left, out bottom_left);
-      getLocalGlobalPoint(1, 1, -1, out local_bottom_right, out bottom_right);
+      switch (GetSelectedInteractionVolume()) {
+        case LeapServiceProvider.InteractionVolumeVisualization.None:
+          break;
+        case LeapServiceProvider.InteractionVolumeVisualization.LeapMotionController:
+          DrawLeapMotionControllerInteractionZone(LMC_BOX_WIDTH, LMC_BOX_DEPTH, LMC_BOX_RADIUS, Color.white);
+          break;
+        case LeapServiceProvider.InteractionVolumeVisualization.Rigel:
+          DrawRigelInteractionZoneMesh();
+          break;
+        case LeapServiceProvider.InteractionVolumeVisualization.Automatic:
+          DetectConnectedDevice();
+          break;
+        default:
+          break;
+      }
+
+    }
+
+    private void ParseRigelInteractionMeshData() {
+
+      if (_rigelInteractionZoneMesh == null) {
+        _rigelInteractionZoneMesh = (Mesh)AssetDatabase.LoadAssetAtPath(Path.Combine("Assets", "UnityModules", "Assets", "Plugins", "LeapMotion", "Core", "Models", "Rigel-interaction-cone-placeholder.obj"), typeof(Mesh));
+
+        // There is a bug/feature where the asset database appears to cache changes to assets, so loading an
+        // asset again (when _rigelInteractionZoneMesh is null) causes it to retain the flipped normals 
+        // it and will then flip them again. There appears to be no way to clear this cache, so here we
+        // check the first normal in the mesh to see if we need to flip it. 
+       // if (_rigelInteractionZoneMesh.normals[0].ApproxEquals(new Vector3(-1.0f, 0.1f, 0)))
+          ReverseNormals();
```

Comment:
This step should not be necessary!  The normals on the mesh are fine when they're imported (as Unity automatically calculates them upon initial import).  See the comment in the shader for more details....